File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/SignatureOutputTests.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.math.BigDecimal;
import org.junit.jupiter.api.Test;

class SignatureOutputTests extends CqlTestBase {

    @Test
    void evaluate() {
        var results = engine.evaluate(toElmIdentifier("SignatureOutputTests"));
        var value = results.forExpression("TestIntegerOverload").value();
        assertThat(value, is(new BigDecimal("1")));

        value = results.forExpression("TestDecimalOverload").value();
        assertThat(value, is(new BigDecimal("1.0")));

        value = results.forExpression("TestMultipleOverload").value();
        assertThat(value, is(5));

        value = results.forExpression("TestIntegerMultipleOverload").value();
        assertThat(value, is(1));

        value = results.forExpression("TestDecimalMultipleOverload").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestIntegerAndDecimalMultipleOverload").value();
        assertThat(value, is(1));
    }
}

File: Src/java/engine/src/test/kotlin/org/opencds/cqf/cql/engine/execution/SignatureOutputTests.kt
package org.opencds.cqf.cql.engine.execution

import java.math.BigDecimal
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class SignatureOutputTests : CqlTestBase() {
    @Test
    fun evaluate() {
        val results = engine.evaluate(toElmIdentifier("SignatureOutputTests"))
        var value = results.forExpression("TestIntegerOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("1")))

        value = results.forExpression("TestDecimalOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("1.0")))

        value = results.forExpression("TestMultipleOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(5))

        value = results.forExpression("TestIntegerMultipleOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(1))

        value = results.forExpression("TestDecimalMultipleOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("2.0")))

        value = results.forExpression("TestIntegerAndDecimalMultipleOverload")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(1))
    }
}
